src/syntaxtree.c

```diff
@@ -194,29 +196,36 @@ static Obj SyntaxTreeDefaultCompiler(Obj result, Expr expr)
         Obj  compiled;
 
         if (comp.args[i].argcomp) {
-            Expr subexpr = READ_EXPR(expr, i);
+            Expr subexpr = READ_EXPR(expr, i + offset);
             compiled = comp.args[i].argcomp(subexpr);
         }
         else {
-            // special case: the last argument may have zero as decompiler,
-            // meaning that all remaining slots of the statement should be
-            // decompiled into a single list
-            const UInt offset = comp.arity - 1;
-            GAP_ASSERT(i == offset);
-
-            // compile the complete rest into one statement
+            // special case: one argument may have zero as decompiler,
+            // which expresses that this slot is "variadic", and
+            // is decompiled into a single list
```

Comment:
Why is this part of the change necessary?

Reply:
We use `ARGS_EXPR` to denote a variadic section of a bag. Previously, the ARGS_EXPR section was always at the end of the bag, but in EXPR_ELM_LIST_LEV, the variadic bit (which represents the dimensions indexed is in the middle of the bag, so I need to support ARGS_EXPR in any position.

While I've already written this code, an alternative option would be to change how we store EXPR_ELM_LIST_LEV expressions (which after investigation seem to be the only type which don't put the variadic bit at the end), to put the variadic bit at the end -- that would obviously break any saved workspaces, but I can't think of any other serious problems it would cause.

Reply:
Nah, it's fine, I just wanted to understand what's going on. Thank you!